downloads/swan-lake-release-notes/swan-lake-2201.0.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,381 @@
+
+<em>Swan Lake 2201.0.0 is the first major release of 2021 and it includes a new set of features and significant improvements to the compiler, runtime, standard library, and developer tooling. It is based on the 2021R1 version of the Language Specification.</em> 
+
+### Updating Ballerina
+
+If you are already using Ballerina, use the [Ballerina Update Tool](/learn/tooling-guide/cli-tools/update-tool/) to directly update to Swan Lake Beta6 by running the command below.
+
+> `bal dist pull 2201.0.0`
+
+### Installing Ballerina
+
+If you have not installed Ballerina, then download the [installers](/downloads/#swanlake) to install.
+
+### Language Updates
+
+#### New Features
+
+##### Support to Refer to Parameters and Local Variables in Object Constructor Expressions
+
+Object constructor expressions can now refer to parameters and local variables. This feature is also available when using an object constructor expression with the `service` qualifier to create a service object. These references can be in field initializers and object methods including the `init` method.
+
+```ballerina
+type Person object {
+   string name;
+   string code;
+  
+   function getRegistrationId(string prefix) returns string;
+};
+ 
+function createPerson(string name, int id, string department) returns Person|error {
+   string codeId = check getCode(id);
+ 
+   Person person = object {
+       string name;
+       string code = codeId; // Refers to the `codeId` local variable.
+ 
+       function init() {
+           self.name = name; // Refers to the `name` parameter.
+       }
+ 
+       function getRegistrationId(string prefix) returns string {
+           string regId = department + ":" + self.code; // Refers to the `department` parameter.
+ 
+           if (prefix.length() > 0) {
+               return prefix + "-" + regId;
+           }
+ 
+           return regId;
+       }
+   };
+ 
+   return person;
+}
+ 
+function getCode(int id) returns string|error {
+   if id < 0 {
+       return error("Invalid ID");
+   }
+   return id.toString();
+}
+```
+
+#### Bug Fixes
+
+- Fixed a bug that caused constants to be resolved with incorrect types in certain scenarios. The type of constants is now the intersection of readonly and the singleton type containing just the shape of the value named by the constant.
+
+  ```ballerina
+  // Previously, the type of X was `int` and now it is singleton `2`.
+  const int X = 1 + 1;
+   
+  public function main() {
+     // Now results in an error.
+     X y = 3;
+  }
+  ```
+
+  ```ballerina
+  // Previously, the type of X was `map<int>` and now it is the intersection of readonly
+  // and the singleton type containing just the mapping value `{a: 1}`.
+  const map<int> X = {a: 1};
+   
+  public function main() {
+     // Now, this results in an error.
+     X y = {a: 2};
+  }
+  ```
+
+- Fixed a spec deviation in type narrowing. This may result in types that were previously narrowed no longer being narrowed
```

Comment:
```suggestion
- Fixed a spec deviation in type narrowing. This may result in types that were previously narrowed no longer being narrowed.
```

Reply:
We didn't use the full stop in other bullet points. Therefore, need to do the same here to preserve consistency.

Reply:
Please check the Standard Library updates section.

Reply:
Lang bullet points have more details, so I think we can make an exception for these.

Reply:
+1